File: Anul 2/MAP/Sem9/seminar9/seminar9/domain/AbstractTaskRunner.cs

using System.ComponentModel;

namespace seminar9.domain
{
    
    public class AbstractTaskRunner: ITaskRunner
    {
        private IContainer container;
        public AbstractTaskRunner(Strategy strategy)
        {
            container=new TaskContainerFactory().CreateContainer(strategy);
        }

        public void addTask(Task t)
        {
            container.Add(t);
        }

        public void executeAll()
        {
            while (!container.IsEmpty())
            {
                executeOneTask();
            }
        }

        public void executeOneTask()
        {
            container.Remove().Execute();
        }

        public bool hasTask()
        {
            return container.Size != 0;
        }
    }
}

File: Anul 2/Semester 1/MAP/Seminar 1/src/com/company/sem1/runner/AbstractTaskRunner.java
package com.company.sem1.runner;

import com.company.sem1.model.Task;
import com.company.sem1.runner.TaskRunner;

public class AbstractTaskRunner implements TaskRunner {
    private TaskRunner runner;

    public AbstractTaskRunner(TaskRunner runner) {
        this.runner= runner;
    }

    @Override
    public void executeOneTask() {
        runner.executeOneTask();
    }

    @Override
    public void executeAll() {
        while(runner.hasTask())
            executeOneTask();
    }

    @Override
    public void addTask(Task t) {
        runner.addTask(t);
    }

    @Override
    public boolean hasTask() {
        return runner.hasTask();
    }
}
